src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,49 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name){
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  die(){
+    Animal.alive = Animal.alive.filter(animal => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name){
+    super(name);
+    this.hidden = false;
+  }
+
+  hide(){
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name){
+    super(name);
+  }
+
+  bite(herbivore){
+    if(!(herbivore instanceof Herbivore)){
+      return;
+    }
+    if(herbivore.hidden){
+      return;
+    }
```

Comment:
```suggestion
  bite(herbivore) {
    if (!(herbivore instanceof Herbivore)) {
      return;
    }

    if (herbivore.hidden) {
      return;
    }
```
Such formatting also might lead to linter errors - fix formatting